core/src/toga/fonts.py

```diff
@@ -35,10 +35,21 @@ def __init__(
         self,
         family: str,
         size: int | str,
+        *,
+        weight: str = NORMAL,
         style: str = NORMAL,
         variant: str = NORMAL,
-        weight: str = NORMAL,
     ):
```

Comment:
Changed the argument order to be consistent with `Font.register`, and with normal English usage (one says "bold italic", not "italic bold". Since the only usage of this constructor in the public API is for drawing text on a Canvas, the impact should be minimal, but I've made the arguments keyword-only to avoid any silent behavior changes.